src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -355,11 +383,19 @@ instance ToDiagnostic FossaError where
         ]
     ConnectionTimeoutError ereq ->
       vsep
-        [ "Connecting to the FOSSA endpoint took too long."
+        [ "The request to the FOSSA endpoint took too long to send."
+        , ""
+        , "This typically means that the CLI is being asked to send too much data"
+        , "with the current network speed (for example, uploading large archives),"
+        , "although this can also be a transient error caused by congested"
+        , "networking conditions between the CLI and the FOSSA API."
+        , ""
+        , "To reduce the likelihood of this error, ensure that only data you really"
+        , "need FOSSA to scan is being uploaded."
```

Comment:
+1 second this

Reply:
I would like to, but:
- At this point without larger refactoring, we're not actually certain that the user is even specifying `vendored-dependencies`, so linking to it seems like it may be confusing as often as it may be helpful.
- We don't have a great way of testing these links to ensure they're durable.

I think we'd ideally link to the documentation like this, but I'd like to consider it out of scope for this PR.